File: java_patterns/src/Basic_Patterns/Behavioural_Patterns/Visitor/Concept/code/Visitors/ConcreteVisitor2.java
package Basic_Patterns.Behavioural_Patterns.Visitor.Concept.code.Visitors;

import Basic_Patterns.Behavioural_Patterns.Visitor.Concept.code.Elements.ConcreteElementA;
import Basic_Patterns.Behavioural_Patterns.Visitor.Concept.code.Elements.ConcreteElementB;

public class ConcreteVisitor2 implements Visitor {
    @Override
    public void visit(ConcreteElementA element) {
        System.out.println(element.exclusiveMethodOfConcreteElementA() + " + ConcreteVisitor2");
    }

    @Override
    public void visit(ConcreteElementB element) {
        System.out.println(element.specialMethodOfConcreteElementB() + " + ConcreteVisitor2");
    }
}

File: src/Basic_Patterns/Behavioural_Patterns/Visitor/Concept/code/visitors/ConcreteVisitor2.ts
import {ConcreteElementA} from "src/Basic_Patterns/Behavioural_Patterns/Visitor/Concept/code/elements/ConcreteElementA";
import {Visitor} from "src/Basic_Patterns/Behavioural_Patterns/Visitor/Concept/code/visitors/Visitor";
import {ConcreteElementB} from "src/Basic_Patterns/Behavioural_Patterns/Visitor/Concept/code/elements/ConcreteElementB";

class ConcreteVisitor2 implements Visitor {
    public visitConcreteElementA(element: ConcreteElementA): void {
        console.log(`${element.exclusiveMethodOfConcreteElementA()} + ConcreteVisitor2`);
    }

    public visitConcreteElementB(element: ConcreteElementB): void {
        console.log(`${element.specialMethodOfConcreteElementB()} + ConcreteVisitor2`);
    }
}

export { ConcreteVisitor2 };